src/features/metadata-based-view/MetadataBasedItemList.js

```diff
@@ -121,23 +163,57 @@ class MetadataBasedItemList extends React.Component<Props, State> {
                 break;
             case FILE_NAME_COLUMN_INDEX:
                 cellData = (
-                    <PlainButton type="button" onClick={() => this.handleOnClick(item)}>
+                    <PlainButton type="button" onClick={() => this.handleItemClick(item)}>
                         {name}
                     </PlainButton>
                 );
                 break;
             default: {
-                const data = getProp(item, 'metadata.data', {});
                 const mdFieldName = this.getMetadataColumnName(metadataColumn);
+                const field = find(fields, ['name', mdFieldName]);
+                if (!field) {
+                    return cellData;
+                }
+                const { type, value, options = [] } = field;
                 cellData = (
                     <>
-                        {data[mdFieldName]}
-                        {isCellEditable && (
-                            <Tooltip text={intl.formatMessage(messages.editLabel)}>
-                                <PlainButton type="button">
-                                    <IconPencil />
-                                </PlainButton>
-                            </Tooltip>
+                        {!isCellBeingEdited && value}
+                        {value && isCellEditable && (
+                            <IconWithTooltip
+                                type={EDIT_ICON_TYPE}
+                                tooltipText={intl.formatMessage(messages.editLabel)}
+                                onClick={() => this.handleEditIconClick(columnIndex, rowIndex)}
+                            />
+                        )}
+                        {value && isCellBeingEdited && (
+                            <div className="bdl-MetadataBasedItemList-cell--edit">
+                                <Field
+                                    canEdit
+                                    dataKey={value}
+                                    dataValue={value}
+                                    displayName=""
```

Comment:
Do we have a value that we can pass in here? This propagates to the text input label which is helpful for leveraging browser autofill capability

Reply:
This scenario occurs because we are using `<Fields />` component from "features/metadata-instance-editor". If you pass any display name, it breaks the table layout we need. This is something we can tackle when eventually `<Fields />` components will . be moved in its own feature and this prop can be made optional.